backend/server.spec.js

```diff
@@ -0,0 +1,77 @@
+import { createTestClient } from 'apollo-server-testing'
+import { gql } from 'apollo-server'
+import Server from './server.js'
+import { InMemoryDataSource, Post } from './db.js'
+
+let db = new InMemoryDataSource()
```

Comment:
You removed my comment `FIXME` but didn't fix the non-atomic test setup:
```suggestion
let db
beforeEach(() => {
  db = new InMemoryDataSource()
})
```